src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageSource.java

```diff
@@ -206,20 +240,36 @@ public KafkaMessageSource(ConsumerFactory<K, V> consumerFactory,
 	 * the records from the previous poll are consumed.
 	 *
 	 * @param consumerFactory the consumer factory.
+	 * @param consumerProperties the consumer properties.
 	 * @param ackCallbackFactory the ack callback factory.
 	 * @param allowMultiFetch true to allow {@code max.poll.records > 1}.
-	 * @param topics the topics.
 	 * @since 3.2
 	 */
 	public KafkaMessageSource(ConsumerFactory<K, V> consumerFactory,
-			KafkaAckCallbackFactory<K, V> ackCallbackFactory, boolean allowMultiFetch, String... topics) {
+			ConsumerProperties consumerProperties,
+			KafkaAckCallbackFactory<K, V> ackCallbackFactory,
+			boolean allowMultiFetch) {
 
 		Assert.notNull(consumerFactory, "'consumerFactory' must not be null");
 		Assert.notNull(ackCallbackFactory, "'ackCallbackFactory' must not be null");
-		Assert.isTrue(topics != null && topics.length > 0, "At least one topic is required");
+		Assert.isTrue(consumerProperties.getTopics() != null && consumerProperties.getTopics().length > 0, "At least one topic is required");
```

Comment:
I hope you are going to address this in your next PR for `topicPattern` support...

Reply:
Yeah, I'll handle that as part of #281